src/Azure/Azure.Quantum.Client/SubmitterFactory.cs

```diff
@@ -117,10 +124,21 @@ public static class SubmitterFactory
         /// <typeparam name="T">The type of the submitter interface.</typeparam>
         /// <returns>The submitter instance.</returns>
         private static T? Submitter<T>(
-            IEnumerable<SubmitterInfo> submitters, string target, IWorkspace workspace, string? storageConnection)
+            IEnumerable<SubmitterInfo> submitters, string target, IWorkspace workspace, string? storageConnection, string targetCapability = "")
             where T : class
         {
-            var constructorInfo = ConstructorInfo(submitters, target);
+            var submitter = submitters.FirstOrDefault(s => s.TargetPattern.IsMatch(target));
+            if (submitter is null)
+            {
+                return null;
+            }
+
+            if (submitter.RequiresTargetCapability && string.IsNullOrWhiteSpace(targetCapability))
+            {
+                return null;
```

Comment:
I'm wondering about the error handling here. There are two ways for this function to fail: either no submitter matches the target string, or the required target capability is not present. But I think if a capability is missing, the entry point driver will still say "no submitter found for the target."

Since the Q# compiler itself provides the capability and there's no command line option, I don't think users would run into this, but it could be confusing if a developer adds/changes a submitter and forgets about the capability.

Reply:
I'm making the following change in `Microsoft.Quantum.EntryPointDriver.Azure.Submit<TIn, TOut>()`. Will that work?
```diff
@@ -92,19 +92,22 @@ namespace Microsoft.Quantum.EntryPointDriver
             if (qirSubmission is null && !(QirSubmitter(settings) is null))
             {
                 DisplayError(
                     $"The target {settings.Target} requires QIR submission, but the project was built without QIR. "
                     + "Please enable QIR generation in the project settings.",
                     null);
             }
             else
             {
-                DisplayError($"No submitters were found for the target {settings.Target}.", null);
+                DisplayError(
+                    $"No submitters were found for the target \"{settings.Target}\" " + 
+                    $"and target capability \"{settings.TargetCapability}\".",
+                    null);
             }
 
             return Task.FromResult(1);
         }
```

Reply:
Yes, I think that works, thanks.